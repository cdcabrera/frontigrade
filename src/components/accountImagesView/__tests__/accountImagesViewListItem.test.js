import React from 'react';
import { mount } from 'enzyme';
import AccountImagesViewListItem from '../accountImagesViewListItem';

describe('AccountImagesViewListItem Component', () => {
  it('should render a component', () => {
    const props = {
      item: {
        arn: 'arn:aws:iam::TEST',
        creation_date: '2018-07-07',
        id: '1',
        images: 1,
        instances: 2,
        name: 'Lorem test',
        openshift_instances: null,
        rhel_instances: 1,
        type: 'aws',
        user_id: 1
      }
    };

    const component = mount(<AccountImagesViewListItem {...props} />);

    expect(component.render()).toMatchSnapshot();
  });

  it('should wrap long names with a on hover title', () => {
    const props = {
      item: {
        arn: 'arn:aws:iam::TEST',
        creation_date: '2018-07-07',
        id: '1',
        images: 1,
        instances: 2,
        name: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse sed sodales eros.',
        openshift_instances: null,
        rhel_instances: 1,
        type: 'aws',
        user_id: 1
      }
    };

    const component = mount(<AccountImagesViewListItem {...props} />);

    expect(component.render()).toMatchSnapshot();
  });
});
