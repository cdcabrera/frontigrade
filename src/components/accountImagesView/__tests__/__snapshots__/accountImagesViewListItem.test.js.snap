// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountImagesViewListItem Component should render 1`] = `
<div
  class="list-group-item list-view-pf-stacked"
>
  <div
    class="list-view-pf-main-info"
  >
    <div
      class="list-view-pf-body"
    >
      <div
        class="list-view-pf-description"
      >
        <div
          class="list-group-item-text"
        >
          <div
            class="cloudmeter-split-description"
          >
            <span
              class="cloudmeter-description-left"
            >
              <div
                class="list-group-item-heading"
              >
                Lorem test
              </div>
            </span>
            <span
              class="cloudmeter-description-right"
            />
          </div>
        </div>
      </div>
      <div
        class="list-view-pf-additional-info"
      >
        <div
          class="list-view-pf-additional-info-item"
        >
          <span
            aria-hidden="true"
            class="pficon pficon-screen cloudmeter-listview-infoitem"
          />
          <strong>
            -
          </strong>
           Instances
        </div>
        <div
          class="list-view-pf-additional-info-item cloudmeter-listview-infoitem"
        >
          <span
            aria-hidden="true"
            class="fa fa-clock-o"
          />
          <strong>
            -
          </strong>
           Hours
        </div>
        <div
          class="list-view-pf-additional-info-item cloudmeter-listview-label cloudmeter-listview-label-hidden"
        >
          <span
            class="label label-warning"
          >
            <abbr
              title="Red Hat Enterprise Linux"
            >
              RHEL
            </abbr>
          </span>
        </div>
        <div
          class="list-view-pf-additional-info-item cloudmeter-listview-label cloudmeter-listview-label-hidden"
        >
          <span
            class="label label-primary"
          >
            <abbr
              title="Red Hat OpenShift Container Platform"
            >
              RHOCP
            </abbr>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
